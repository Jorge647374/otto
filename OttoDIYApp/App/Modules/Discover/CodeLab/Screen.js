import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  InstructionList,
  CodeLabNav,
  Modal } from 'App/Components'

import { Metrics } from 'App/Themes'

import Types from 'App/Services/PropTypes'

export default class Screen extends Component {
  static propTypes = {
    config: PropTypes.any,
    instructions: PropTypes.arrayOf(PropTypes.shape({
      ...Types.instruction
    })).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    navHeight: PropTypes.number.isRequired,
    onConnect: PropTypes.func.isRequired,
    onSlidingComplete: PropTypes.func.isRequired,
    onChangeOrder: PropTypes.func.isRequired,
    onNavHeightChange: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    onNavPress: PropTypes.func.isRequired,
    onRun: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  resolveSkills = () => {
    const { config } = this.props
    const { moves, skills } = config || {}

    const allSkills = []
    if (moves && moves.length > 0) {
      allSkills.push({
        id: 'moves',
        category: 'Moves',
        items: moves
      })
    }
    if (skills && skills.length > 0) {
      allSkills.push(...skills)
    }

    return allSkills
  }

  render () {
    const {
      instructions,
      showNotConnectedModal,
      navHeight,
      onConnect,
      onSlidingComplete,
      onChangeOrder,
      onNavHeightChange,
      onClose,
      onNavPress,
      onHideNotConnectedModal} = this.props

    const skills = this.resolveSkills()

    return (
      <Container>
        <InstructionList
          style={{marginBottom: navHeight + Metrics.unit}}
          instructions={instructions}
          onSlidingComplete={onSlidingComplete}
          onChangeOrder={onChangeOrder}
          onClose={onClose} />
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onConnect}
          template='NotConnected' />
        <CodeLabNav
          skills={skills}
          onPress={onNavPress}
          onNavHeightChange={onNavHeightChange} />
      </Container>
    )
  }
}
