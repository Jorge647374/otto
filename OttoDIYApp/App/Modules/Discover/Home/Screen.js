import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  FeaturedListItem,
  CompactListItem } from 'App/Components'

import { Metrics, Colors, Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List title='Featured' cols={2}>
            <FeaturedListItem
              image={Images.featured.led}
              title='Blink'
              onPress={() => { onPress('Blink') }} />
            <FeaturedListItem
              icon={'heartbeat'}
              iconColor={Colors.red}
              title='Heart Beat'
              onPress={() => { onPress('Heart Beat') }} />
          </List>
          <List title='Projects'>
            <CompactListItem
              icon='circle-o'
              title='Hello World'
              onPress={() => { onPress('Hello World') }} />
            <CompactListItem
              icon='circle-o'
              title='Dance Show'
              onPress={() => { onPress('Dance Show') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='DISCOVER'
            text={'Get inspired by what you can create. Build your own creations or remix others to jumpstart your ideas.'}
            button='Code Lab'
            textAlign='left'
            onPress={() => { onPress('Code Lab') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
