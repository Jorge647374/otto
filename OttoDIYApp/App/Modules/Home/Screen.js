import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, List, ListItem, BottomNav } from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    onNavigatePress: PropTypes.func.isRequired
  }

  render () {
    const { onNavigatePress } = this.props
    return (
      <Container>
        <List>
          <ListItem
            title='Play & Explore'
            text='Let’s play'
            button='Play'
            onPress={() => { onNavigatePress('PlayerScreen') }} />
          <ListItem
            title='Discover'
            text='Let’s see what you can do'
            button='Discover'
            onPress={() => { onNavigatePress('DiscoverScreen') }} />
          <ListItem
            title='Learn'
            text='You’ll be a guru in no time'
            button='Learn'
            onPress={() => { onNavigatePress('LearnScreen') }} />
          <ListItem
            title='Lab'
            text='Get under the hood'
            button='Lab'
            onPress={() => { onNavigatePress('LabScreen') }} />
        </List>
        <BottomNav
          onRatePress={() => { onNavigatePress('RateScreen') }}
          onHomePress={() => {}}
          onSettingsPress={() => { onNavigatePress('SettingsScreen') }} />
      </Container>
    )
  }
}
