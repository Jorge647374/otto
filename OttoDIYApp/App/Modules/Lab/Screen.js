import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Links,
  Link,
  List,
  StatListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <List style={{ margin: Metrics.unit * 2 }} cols={2}>
          <StatListItem icon='bluetooth' title='bluetooth' value='Connected' />
          <StatListItem icon='battery' title='battery' value='85%' />
          <StatListItem icon='play' title='playtime' value='4.8h' />
          <StatListItem icon='code' title='remixes' value='5' />
          <StatListItem icon='book' title='lessons' value='2' />
          <StatListItem icon='gamepad' title='level' value='1' />
        </List>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer>
          <Links>
            <Link theme='light' text='Shop' onPress={onPress} />
            <Link theme='light' text='Customize' onPress={onPress} />
            <Link theme='light' text='Add a buddy' onPress={onPress} />
            <Link theme='light' text='View logs' onPress={onPress} />
          </Links>
        </Footer>
      </Container>
    )
  }
}
