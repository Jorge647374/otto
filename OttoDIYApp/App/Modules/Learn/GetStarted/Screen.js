import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    connected: PropTypes.bool.isRequired,
    onDone: PropTypes.func.isRequired,
    onConnectPress: PropTypes.func.isRequired
  }

  render () {
    const { connected, onDone, onConnectPress } = this.props

    const connectButton = (!connected)
      ? {title: 'Connect', onPress: onConnectPress}
      : {title: 'Next', onPress: null}

    const connectCard = (!connected)
      ? {title: 'Get connected', text: 'Letâ€™s get you connected & ready to start playing'}
      : {title: 'Nice work, you`re connected', text: null}

    return (
      <Container>
        <Carousel
          buttons={['Next', connectButton.title, 'Done']}
          onPress={[null, connectButton.onPress, null]}
          onDone={onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.bluetooth}
            title={connectCard.title}
            text={connectCard.text} />
          <Card
            image={Images.done}
            title='Woot woot ðŸŽ‰'
            text='Youâ€™re all setup & ready to play' />
        </Carousel>
      </Container>
    )
  }
}
