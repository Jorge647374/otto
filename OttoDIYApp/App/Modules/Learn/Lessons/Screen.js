import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  ListHeader,
  CompactListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List>
            <ListHeader title='Lessons' />
            <CompactListItem
              icon='circle-o'
              title='Get Started'
              onPress={() => { onPress('Get Started') }} />
            <CompactListItem
              icon='circle-o'
              title='Play Time'
              onPress={() => { onPress('Play Time') }} />
            <CompactListItem
              icon='circle-o'
              title='Customize'
              onPress={() => { onPress('Customize') }} />
            <CompactListItem
              icon='circle-o'
              title='Code Remix'
              onPress={() => { onPress('Code Remix') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='START YOUR NEXT LESSON'
            text={'Learn them skills!'}
            button='Start'
            textAlign='left'
            onPress={() => { onPress('Get Started') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
