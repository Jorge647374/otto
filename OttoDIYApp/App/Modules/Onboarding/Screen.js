import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card
} from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired
  }

  render () {
    const { onDone } = this.props

    return (
      <Container>
        <Carousel
          buttons={['Next', 'Next', 'Done']}
          onPress={[null, null, null]}
          onDone={onDone}>
          <Card
            image={Images.onboarding.one}
            title='Welcome to the Otto DIY App'
            text='Where you can learn coding and bring your robot to live'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.two}
            title='Otto awaits your commands'
            text='Otto DIY+, Eyes, Wheels & Humanoid supported out of the box!'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.three}
            title='Lots more to come! ðŸŽ‰'
            text='Games, more robots, customize your Otto and more fun!'
            style={{titleViewText: {fontSize: 22}}}
          />
        </Carousel>
      </Container>
    )
  }
}
