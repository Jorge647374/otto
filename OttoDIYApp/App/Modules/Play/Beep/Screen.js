import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Types from 'App/Services/PropTypes'

import {
  Container,
  List,
  SquareListItem,
  Modal} from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    sounds: PropTypes.arrayOf(Types.sound).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    onPlay: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired,
    onBack: PropTypes.func.isRequired
  }

  render () {
    const {
      sounds,
      showNotConnectedModal,
      onPlay,
      onHideNotConnectedModal,
      onBack
    } = this.props
    return (
      <Container>
        <List cols={2} scrollable>
          {sounds.map(sound => {
            return (
              <SquareListItem
                key={sound.key}
                icon='play'
                iconStyle={{marginLeft: 4}}
                title={sound.name}
                onPress={() => { onPlay(sound) }} />
            )
          })}
        </List>
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onBack}
          template='NotConnected' />
      </Container>
    )
  }
}
