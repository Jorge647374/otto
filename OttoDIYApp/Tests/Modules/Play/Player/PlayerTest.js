import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Play/Player/Screen'
import Config from 'App/Services/Client/Otto/Config'

describe('Player', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen
        navigation={{navigate: () => {}}}
        speed='medium'
        connected
        config={Config}
        showNotConnectedModal={false}
        onConnect={() => {}}
        onUp={() => {}}
        onDown={() => {}}
        onLeft={() => {}}
        onRight={() => {}}
        onupdown={() => {}}
        onLongPress={() => {}}
        onLongPressOut={() => {}}
        onSkillPress={() => {}}
        onToggleSpeed={() => {}}
        onHelp={() => {}}
        onHideNotConnectedModal={() => {}}
        onChooseRobotPress={() => {}}
        onCodeLabPress={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
