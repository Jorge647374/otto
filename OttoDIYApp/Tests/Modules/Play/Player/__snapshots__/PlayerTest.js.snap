// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component renders correctly 1`] = `
<Container
  dark={true}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 16,
        "marginBottom": 0,
        "marginTop": 32,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
      />
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ChamferImageButton
      image={
        Object {
          "process": [Function],
        }
      }
      onPress={[Function]}
    />
    <View
      style={
        Object {
          "marginHorizontal": 16,
        }
      }
    >
      <ArrowJoystick
        onDown={[Function]}
        onLeft={[Function]}
        onLongPress={[Function]}
        onLongPressOut={[Function]}
        onRight={[Function]}
        onUp={[Function]}
      />
    </View>
    <ChamferImageButton
      image={
        Object {
          "process": [Function],
        }
      }
      onPress={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 16,
      }
    }
  >
    <View
      key="0"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        key="1"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="2"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="3"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="4"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="5"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
    <View
      key="6"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        key="7"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="8"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="9"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="10"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="11"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
    <View
      key="12"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        key="13"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="14"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="15"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="16"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        key="17"
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#2D2D64",
          "borderTopColor": "#FAFAFA",
          "borderTopWidth": 1,
          "bottom": 0,
          "left": 0,
          "minHeight": 80,
          "paddingBottom": 32,
          "paddingTop": 32,
          "position": "absolute",
          "width": "100%",
          "zIndex": 9999,
        },
        Object {
          "borderTopWidth": 0,
        },
      ]
    }
  >
    <BottomNav
      onTabPress={[Function]}
      tabs={
        Array [
          Object {
            "category": "Skills",
            "id": "skills",
            "image": Object {
              "process": [Function],
            },
            "items": Array [
              Object {
                "cmd": "M 19",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Jitter",
              },
              Object {
                "cmd": "M 8",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Swing",
              },
              Object {
                "cmd": "M 14",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Tiptoe",
              },
              Object {
                "cmd": "M 20",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Ascend",
              },
              Object {
                "cmd": "M 5",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Up Down",
              },
              Object {
                "cmd": "M 13",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Flap Back",
              },
              Object {
                "cmd": "M 7",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Moonwalk Left",
              },
              Object {
                "cmd": "M 17",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Shake Left",
              },
              Object {
                "cmd": "M 15",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Bend Left",
              },
              Object {
                "cmd": "M 10",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Cross Left",
              },
              Object {
                "cmd": "M 12",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Flap Front",
              },
              Object {
                "cmd": "M 6",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Moonwalk Right",
              },
              Object {
                "cmd": "M 18",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Shake Right",
              },
              Object {
                "cmd": "M 16",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Bend Right",
              },
              Object {
                "cmd": "M 9",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": true,
                "title": "Cross Right",
              },
            ],
          },
          Object {
            "category": "Sounds",
            "id": "sounds",
            "image": Object {
              "process": [Function],
            },
            "items": Array [
              Object {
                "cmd": "K 16",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sing 1",
              },
              Object {
                "cmd": "K 17",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sing 2",
              },
              Object {
                "cmd": "K 18",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sing 3",
              },
              Object {
                "cmd": "K 4",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "OhOoh",
              },
              Object {
                "cmd": "K 19",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Push",
              },
              Object {
                "cmd": "K 10",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Kinda Happy",
              },
              Object {
                "cmd": "K 8",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Happy",
              },
              Object {
                "cmd": "K 9",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Super Happy",
              },
              Object {
                "cmd": "K 11",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sad",
              },
              Object {
                "cmd": "K 7",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sleeping",
              },
              Object {
                "cmd": "K 3",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Surprise",
              },
              Object {
                "cmd": "K 2",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Disconnection",
              },
              Object {
                "cmd": "K 12",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Confused",
              },
              Object {
                "cmd": "K 15",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Fart",
              },
              Object {
                "cmd": "K 6",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Cuddly",
              },
            ],
          },
          Object {
            "category": "Gestures",
            "id": "gestures",
            "image": Object {
              "process": [Function],
            },
            "items": Array [
              Object {
                "cmd": "H 1",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Happy",
              },
              Object {
                "cmd": "H 2",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Super Happy",
              },
              Object {
                "cmd": "H 7",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Love",
              },
              Object {
                "cmd": "H 3",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sad",
              },
              Object {
                "cmd": "H 13",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Fail",
              },
              Object {
                "cmd": "H 6",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Confused",
              },
              Object {
                "cmd": "H 5",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Fart",
              },
              Object {
                "cmd": "H 9",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Fretful",
              },
              Object {
                "cmd": "H 8",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Angry",
              },
              Object {
                "cmd": "H 4",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Sleeping",
              },
              Object {
                "cmd": "H 10",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Magic",
              },
              Object {
                "cmd": "H 12",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Victory",
              },
              Object {
                "cmd": "H 11",
                "image": Object {
                  "process": [Function],
                },
                "stopAtEnd": false,
                "title": "Wave",
              },
            ],
          },
        ]
      }
    />
  </View>
  <Modal
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    onBack={[Function]}
    onHidePress={[Function]}
    show={false}
    template="NotConnected"
  />
</Container>
`;
